src/txn/txn_timestamp.c

```diff
@@ -854,50 +854,67 @@ __wt_txn_set_read_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t read_ts)
 int
 __wt_txn_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
 {
-    WT_CONFIG_ITEM cval;
+    WT_CONFIG cparser;
+    WT_CONFIG_ITEM ckey, cval;
     WT_DECL_RET;
-    wt_timestamp_t ts;
+    wt_timestamp_t commit_ts, durable_ts, prepare_ts, read_ts;
     bool set_ts;
 
     set_ts = false;
+    commit_ts = durable_ts = prepare_ts = read_ts = 0;
+
     WT_TRET(__wt_txn_context_check(session, true));
 
-    /* Look for a commit timestamp. */
-    ret = __wt_config_gets_def(session, cfg, "commit_timestamp", 0, &cval);
-    WT_RET_NOTFOUND_OK(ret);
-    if (ret == 0 && cval.len != 0) {
-        WT_RET(__wt_txn_parse_timestamp(session, "commit", &ts, &cval));
-        WT_RET(__wt_txn_set_commit_timestamp(session, ts));
-        set_ts = true;
+    /*
+     * If the API received no configuration string, there's nothing to do.
+     */
+    if (cfg == NULL || cfg[0] == NULL || cfg[1] == NULL)
+        return (0);
+
+    /*
+     * We take a shortcut in parsing that works because we're only given a base configuration and a
+     * user configuration.
+     */
+    WT_ASSERT(session, cfg[0] != NULL && cfg[1] != NULL && cfg[2] == NULL);
+    __wt_config_init(session, &cparser, cfg[1]);
+    while ((ret = __wt_config_next(&cparser, &ckey, &cval)) == 0) {
+        WT_ASSERT(session, ckey.str != NULL);
+        if (WT_STRING_MATCH("commit_timestamp", ckey.str, ckey.len)) {
+            WT_RET(__wt_txn_parse_timestamp(session, "commit", &commit_ts, &cval));
+            set_ts = true;
+        } else if (WT_STRING_MATCH("durable_timestamp", ckey.str, ckey.len)) {
+            WT_RET(__wt_txn_parse_timestamp(session, "durable", &durable_ts, &cval));
+            set_ts = true;
```

Comment:
I want to call out the setting of the `set_ts` variable when `durable_timestamp` is specified and also when `prepare_timestamp` is specified.  The original code does not set this for these cases, and that seems incorrect to me, and so is "fixed" here.  As I understand it, having this be true buts an informational message in the log (Added in WT-4712).  So we could potentially be missing some of these informational messages.  Though it's possible MongoDB never triggers this problem depending on how it uses the API.

@sueloverso, you made the original change, please let me know if I misunderstand this.